From zaitcev@redhat.com  Tue Aug 14 22:16:07 2007
From: Pete Zaitcev <zaitcev@redhat.com>
Date: Tue, 14 Aug 2007 00:42:53 -0700
Subject: usbmon: Smooth the core code
To: greg@kroah.com
Cc: zaitcev@redhat.com, linux-usb-devel@lists.sourceforge.net
Message-ID: <20070814004253.9bc5bce8.zaitcev@redhat.com>


Two things:
 - mbus can be NULL (in case of bus removal while reader is reading)
 - Remove a useless assignment

Signed-off-by: Pete Zaitcev <zaitcev@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/mon/mon_main.c |   16 +++-------------
 1 file changed, 3 insertions(+), 13 deletions(-)

--- a/drivers/usb/mon/mon_main.c
+++ b/drivers/usb/mon/mon_main.c
@@ -148,18 +148,8 @@ static void mon_complete(struct usb_bus 
 {
 	struct mon_bus *mbus;
 
-	mbus = ubus->mon_bus;
-	if (mbus == NULL) {
-		/*
-		 * This should not happen.
-		 * At this point we do not even know the bus number...
-		 */
-		printk(KERN_ERR TAG ": Null mon bus in URB, address %p\n",
-		    urb);
-		return;
-	}
-
-	mon_bus_complete(mbus, urb);
+	if ((mbus = ubus->mon_bus) != NULL)
+		mon_bus_complete(mbus, urb);
 	mon_bus_complete(&mon_bus0, urb);
 }
 
@@ -170,7 +160,7 @@ static void mon_complete(struct usb_bus 
  */
 static void mon_stop(struct mon_bus *mbus)
 {
-	struct usb_bus *ubus = mbus->u_bus;
+	struct usb_bus *ubus;
 	struct list_head *p;
 
 	if (mbus == &mon_bus0) {
